apps/els_lsp/src/els_hover_provider.erl

```diff
@@ -23,24 +31,72 @@
 is_enabled() ->
   true.
 
+-spec init() -> state().
+init() ->
+  #{ in_progress => []}.
+
 -spec handle_request(any(), state()) -> {any(), state()}.
 handle_request({hover, Params}, State) ->
+  #{in_progress := InProgress} = State,
   #{ <<"position">>     := #{ <<"line">>      := Line
                             , <<"character">> := Character
                             }
    , <<"textDocument">> := #{<<"uri">> := Uri}
    } = Params,
+   ?LOG_DEBUG("Starting hover job ""[uri=~p, line=~p, character=~p]"
+             , [Uri, Line, Character]
+             ),
+   Job = run_hover_job(Uri, Line, Character),
+   {Job, State#{in_progress => [{Uri, Job}|InProgress]}}.
+
+
+-spec handle_info(any(), state()) -> state().
+handle_info({result, HoverResp, Job}, State) ->
+  ?LOG_DEBUG("Received hover result [job=~p]", [Job]),
+  #{ in_progress := InProgress } = State,
+  els_server:send_response(Job, HoverResp),
+  State#{ in_progress => lists:keydelete(Job, 2, InProgress) }.
+
+-spec cancel_request(job(), state()) -> state().
+cancel_request(Job, State) ->
+  ?LOG_DEBUG("Cancelling hover [job=~p]", [Job]),
+  els_background_job:stop(Job),
+  #{ in_progress := InProgress } = State,
+  State#{ in_progress => lists:keydelete(Job, 2, InProgress) }.
+
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+
+-spec run_hover_job(uri(), line(), column()) -> pid().
+run_hover_job(Uri, Line, Character) ->
   {ok, Doc} = els_utils:lookup_document(Uri),
   POIs = els_dt_document:get_element_at_pos(Doc, Line + 1, Character + 1),
```

Comment:
I assume you've measured that these operations are fast? They could be moved into the background job as well.

Reply:
Hoestly, it's a copy paste artifact. Let's move them into the background job

Reply:
I moved it into the background job